File: src/main/Python/boj_step13/boj10814.py
import sys
cin = sys.stdin.readline

class Member:
    def __init__(self, id, age, name):
        self.id = id
        self.age = age
        self.name = name

N = int(cin())
L = []
for id in range(N):
    age, name = cin().split()
    age = int(age)
    L.append(Member(id + 1, age, name))
L.sort(key=lambda x: (x.age, x.id))
for info in L: print(f'{info.age} {info.name}')

File: src/main/Cpp/boj_step13/boj10814.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct DATA {
    int id{};
    int age{};
    string name;
};

int main() {
    cin.tie(nullptr);
    ios_base::sync_with_stdio(false);

    int N;
    cin >> N;
    vector<DATA> list(N);
    int id{1};

    for (auto& member: list) {
        int age;
        string name;
        cin >> age >> name;
        member.id = id++;
        member.age = age;
        member.name = name;
    }

    sort (list.begin(), list.end(), [] (DATA a, DATA b) {
        if (a.age == b.age)
            return a.id < b.id;

        return a.age < b.age;
    });

    for (const auto& member: list)
        cout << member.age << ' ' << member.name << '\n';
}